plugins/node/opentelemetry-instrumentation-redis-4/src/instrumentation.ts

```diff
@@ -332,6 +335,48 @@ export class RedisInstrumentation extends InstrumentationBase<any> {
     };
   }
 
+  private _getPatchedClientConnect() {
+    const plugin = this;
+    return function connectWrapper(original: Function) {
+      return function patchedConnect(this: any): Promise<void> {
+        const options = this.options;
+
+        const attributes = getClientAttributes(options);
+
+        const span = plugin.tracer.startSpan(
+          `${RedisInstrumentation.COMPONENT}-connect`,
+          {
+            kind: SpanKind.CLIENT,
+            attributes,
+          }
+        );
+
+        const res = context.with(trace.setSpan(context.active(), span), () => {
+          return original.apply(this);
+        });
+
+        return res
+          .then((result: unknown) => {
+            return new Promise(resolve => {
+              span.end();
+              resolve(result);
+            });
```

Comment:
This doesn't need the use of the Promise constructor:
```suggestion
            span.end();
            return result;
```

Reply:
Fixed :pray: